tuf/client/updater.py

```diff
@@ -1130,28 +1131,62 @@ def _update_root_metadata(self, current_root_metadata):
       None.
     """
 
-    # Retrieve the latest, remote root.json.
-    latest_root_metadata_file = self._get_metadata_file(
-        'root', 'root.json', DEFAULT_ROOT_UPPERLENGTH, None)
+    def neither_403_nor_404(mirror_error):
+      WHITELIST = {403, 404}
+      if isinstance(mirror_error, six.moves.urllib.error.HTTPError):
+        if mirror_error.code in WHITELIST:
+          return False
+      elif isinstance(mirror_error, requests.exceptions.HTTPError):
```

Comment:
Do you expect `requests.exceptions.HTTPError` here? I think you should be fine with  `six.moves.urllib.error.HTTPError`.

Reply:
Apparently we do. For some reason I haven't figured out yet, I see the first exception on my local tox, but the second one of the Travis tox. Makes zero sense, but that's what seems to be happening. I'll try removing either one and check on both environments.

Reply:
I just cannot reproduce what you're saying. I ran Travis on [a commit that re-raises the exception if it is an instance of `requests.exceptions.HTTPError`](https://github.com/lukpueh/tuf/blob/70d9146909019ebe523bfb199d3a6f2d0056a52f/tuf/client/updater.py#L1139-L1141), but the exception is never raised.

If you scroll down to the end of the corresponding Travis build logs and take a look at the coverage summary you can see how the corresponding lines ([L1140 - L1141](https://github.com/lukpueh/tuf/blob/70d9146909019ebe523bfb199d3a6f2d0056a52f/tuf/client/updater.py#L1140-L1141)) never get touched (Travis logs for [Py2](https://travis-ci.com/lukpueh/tuf/jobs/210722135), [Py3.4](https://travis-ci.com/lukpueh/tuf/jobs/210722136) and [Py3.6](https://travis-ci.com/lukpueh/tuf/jobs/210722137)).

Reply:
Btw. I don't understand what's going on in the coverage reports in your latest Travis builds. It shows missing branches that correspond to docstring lines. Something seems to be off there.

Reply:
It doesn't seem likely, but could this somehow be related to TUF's Travis builds haven't yet been [migrated to `travis-ci.com`](https://docs.travis-ci.com/user/migrate/open-source-on-travis-ci-com/)?

Reply:
Okay, @trishankatdatadog, hold on. I know what happened, at least why I couldn't reproduce the `requests.exceptions.HTTPError` neither locally nor on Travis, and why you only saw them on Travis.

Your Travis builds ran over the PR, i.e. your changes on top of the current upstream `develop` branch. While locally, I suppose, you ran the tests on your `trishankatdatadog:trishankatdatadog/correctly-rotate-root` branch, which is on top of a pretty outdated `develop` branch. I'm sure that if you rebase onto `upstream/develop`, you'll get the same results locally. At least that (an outdated base develop) was the reason why I couldn't reproduce the errors.

Reply:
So it turns out that we can scrap the first instance check, i.e. for `six.moves.urllib.error.HTTPError`. 

`six.moves.urllib.request` was replaced by `requests` in b9bc8602c2bdc31f09b241f3612f7fda6d601ea9. But those changes aren't part of the `trishankatdatadog:trishankatdatadog/correctly-rotate-root` branch, hence the different results locally and on Travis.

Reply:
Great job, @lukpueh, done, thanks!